pkg/operator/operator.go

```diff
@@ -130,7 +130,7 @@ func New(config operatorconfig.Config, kubeConfig *rest.Config) (*Operator, erro
 		return nil, err
 	}
 	// example of future featuregate read and usage to set a variable to pass to a controller
-	_ = sets.NewString(enabled...).Has("AzureWorkloadIdentity")
+	AzureWorkloadIdentityEnabled := sets.NewString(enabled...).Has("AzureWorkloadIdentity")
```

Comment:
You need to check if the feature is disabled, if it's disabled explicitly, then it shouldn't be enabled, even if it's in the enabled list IIUC

Reply:
We discussed in slack, but, to close this off, lets get the PR updated with the latest library-go changes and this becomes very obvious, we are now only allowing to check if something is enabled, the mechanics internally mean you won't need to worry here